src/include/cursor.i

```diff
@@ -210,8 +211,19 @@ __cursor_reset(WT_CURSOR_BTREE *cbt)
     /*
      * If we were scanning and saw a lot of deleted records on this page, try to evict the page when
      * we release it.
+     *
+     * A visible stop timestamp could have been treated as a tombstone and accounted in the deleted
+     * count. Such a page might not have any new updates and be clean, but could benefit from
+     * reconciliation getting rid of the obsolete content. Hence mark the page dirty to force it
+     * through reconciliation.
      */
+    page = cbt->ref->page;
     if (cbt->page_deleted_count > WT_BTREE_DELETE_THRESHOLD) {
+        if (page != NULL) {
```

Comment:
I'm surprised the page can be NULL here - did you see that it was, or were you just being safe?

Reply:
We check for page not being null in `__wt_btcur_next` and `__wt_btcur_next`, so I thought there is a possibility that reset might see a NULL page too.

Reply:
Actually evict soon dereferences page without checking, so page has to exist. I have cleaned it up.